### ISSUE TITLE
How to set data attributes?

### ISSUE THREAD

**Issue by clsource** [mudgen/webscript#10]
Hello, How can the data attributes be set?

https://www.w3schools.com/tags/att_data-.asp

example 
```html
 <ul>
  <li data-animal-type="bird">Owl</li>
  <li data-animal-type="fish">Salmon</li>
  <li data-animal-type="spider">Tarantula</li>
</ul> 
```
Since `-` is considered "minus" something like this is invalid.

```js
 ul(
   li.data-animal-type("bird")("Owl"),
   li.data-animal-type("fish")("Salmon"),
   li.data-animal-type("spider")("Tarantula")
)
```

Thanks 👍

**Comment by mudgen** [comment]
Here is one way to handle it:
```javascript
 ul(
   li["data-animal-type"]`bird`("Owl"),
   li["data-animal-type"]`fish`("Salmon"),
   li["data-animal-type"]`spider`("Tarantula")
)
```
I am thinking about a nicer way to handle this case.

**Comment by clsource** [comment]
One idea maybe having a special use case function
`h(name, value)`

```js
li.h("data-animal-type", "bird")
```

**Comment by mudgen** [comment]
I am not done working on this.

**Comment by mudgen** [comment]
Hey, so I came up with an interesting solution to this.

One way to handle this is to use a different character other than a hyphen that is valid in Javascript property names.  Then automatically replace this character with a hyphen within the Webscript builder logic.

But what character to use in place of a hyphen?  I went on a character search. We could use `$` for this but it does not make a good word separator and I fear that there could be Javascript properties that have `$` in them already.

I found a good character to use that is valid in object properties.  It is called interpunct or the middle dot: https://en.wikipedia.org/wiki/Interpunct

More info and an example in the release: https://github.com/mudgen/webscript/releases/tag/v0.1.1
